### ISSUE TITLE
UseNativeHttpHandler=true is not trimming correctly on mobile apps

### ISSUE THREAD

**Issue by eerhardt** [dotnet/runtime#64361]
In a default Maui Android app, I see the following assemblies present in my Release .apk:

![image](https://user-images.githubusercontent.com/8291187/151270053-f18f45a7-b47e-4f0e-a66b-543c45b76c3a.png)

The first thing that caught my eye was that `System.Net.Quic.dll` is there. But on Mobile apps, we are using `UseNativeHttpHandler=true`, see this [Android SDK setting](https://github.com/xamarin/xamarin-android/blob/60e983c6161242e0322e64c52d113e8cef2d61aa/src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.DefaultProperties.targets#L94).

Since we are using the native HTTP handler, there should be no reason we are bringing .NET sockets handler (and with it HTTP 2 & 3 support).

The reason this code isn't getting trimmed is because we only have a Linker substitution for when `false` is specified.

https://github.com/dotnet/runtime/blob/c0d481fecb23e166d1eff180ed3b20fcbe402875/src/libraries/System.Net.Http/src/ILLink/ILLink.Substitutions.mobile.xml#L1-L7

We should add a substitution entry for `true` as well. This way a bunch of System.Net.* IL can be trimmed from mobile apps by default.

cc @steveisok @jonathanpeppers

**Comment by jonathanpeppers** [comment]
Ok, I added:
```
var foo = await new System.Net.Http.HttpClient().GetStringAsync ("https://google.com");
```
And now I get:

```
1981-01-01 01:01:02 .....        60054        60054  assemblies\Java.Interop.dll
1981-01-01 01:01:02 .....         3677         3677  assemblies\Microsoft.Win32.Primitives.dll
1981-01-01 01:01:02 .....       132888       132888  assemblies\Mono.Android.dll
1981-01-01 01:01:02 .....         6851         6851  assemblies\System.Collections.NonGeneric.dll
1981-01-01 01:01:02 .....         3034         3034  assemblies\System.Diagnostics.DiagnosticSource.dll
1981-01-01 01:01:02 .....        26850        26850  assemblies\System.Formats.Asn1.dll
1981-01-01 01:01:02 .....        11334        11334  assemblies\System.IO.Compression.Brotli.dll
1981-01-01 01:01:02 .....        18508        18508  assemblies\System.IO.Compression.dll
1981-01-01 01:01:02 .....        10369        10369  assemblies\System.Linq.dll
1981-01-01 01:01:02 .....       221610       221610  assemblies\System.Net.Http.dll
1981-01-01 01:01:02 .....        13159        13159  assemblies\System.Net.NameResolution.dll
1981-01-01 01:01:02 .....        17540        17540  assemblies\System.Net.NetworkInformation.dll
1981-01-01 01:01:02 .....        40562        40562  assemblies\System.Net.Primitives.dll
1981-01-01 01:01:02 .....        47907        47907  assemblies\System.Net.Quic.dll
1981-01-01 01:01:02 .....         3627         3627  assemblies\System.Net.Requests.dll
1981-01-01 01:01:02 .....        57785        57785  assemblies\System.Net.Security.dll
1981-01-01 01:01:02 .....        54520        54520  assemblies\System.Net.Sockets.dll
1981-01-01 01:01:02 .....         1285         1285  assemblies\System.Runtime.CompilerServices.Unsafe.dll
1981-01-01 01:01:02 .....         2912         2912  assemblies\System.Runtime.InteropServices.RuntimeInformation.dll
1981-01-01 01:01:02 .....        21292        21292  assemblies\System.Runtime.Numerics.dll
1981-01-01 01:01:02 .....         2519         2519  assemblies\System.Runtime.dll
1981-01-01 01:01:02 .....        42018        42018  assemblies\System.Security.Cryptography.Algorithms.dll
1981-01-01 01:01:02 .....        13373        13373  assemblies\System.Security.Cryptography.Encoding.dll
1981-01-01 01:01:02 .....         8577         8577  assemblies\System.Security.Cryptography.Primitives.dll
1981-01-01 01:01:02 .....        76491        76491  assemblies\System.Security.Cryptography.X509Certificates.dll
1981-01-01 01:01:02 .....        16762        16762  assemblies\System.Threading.Channels.dll
1981-01-01 01:01:02 .....         9733         9733  assemblies\arm64-v8a\System.Collections.Concurrent.dll
1981-01-01 01:01:02 .....         7284         7284  assemblies\arm64-v8a\System.Collections.dll
1981-01-01 01:01:02 .....       609354       609354  assemblies\arm64-v8a\System.Private.CoreLib.dll
1981-01-01 01:01:02 .....        38431        38431  assemblies\arm64-v8a\System.Private.Uri.dll
1981-01-01 01:01:02 .....         9683         9683  assemblies\armeabi-v7a\System.Collections.Concurrent.dll
1981-01-01 01:01:02 .....         6924         6924  assemblies\armeabi-v7a\System.Collections.dll
1981-01-01 01:01:02 .....       604114       604114  assemblies\armeabi-v7a\System.Private.CoreLib.dll
1981-01-01 01:01:02 .....        38816        38816  assemblies\armeabi-v7a\System.Private.Uri.dll
1981-01-01 01:01:02 .....         3967         3967  assemblies\foo.dll
1981-01-01 01:01:02 .....         9683         9683  assemblies\x86\System.Collections.Concurrent.dll
1981-01-01 01:01:02 .....         6924         6924  assemblies\x86\System.Collections.dll
1981-01-01 01:01:02 .....       604218       604218  assemblies\x86\System.Private.CoreLib.dll
1981-01-01 01:01:02 .....        38817        38817  assemblies\x86\System.Private.Uri.dll
1981-01-01 01:01:02 .....         9733         9733  assemblies\x86_64\System.Collections.Concurrent.dll
1981-01-01 01:01:02 .....         7542         7542  assemblies\x86_64\System.Collections.dll
1981-01-01 01:01:02 .....       614678       614678  assemblies\x86_64\System.Private.CoreLib.dll
1981-01-01 01:01:02 .....        38432        38432  assemblies\x86_64\System.Private.Uri.dll
```

The MAUI template would use `HttpClient` because it isn't fully trimmed, and there is support for images from the web like `<Image Source="https://someurl" />`.

**Comment by eerhardt** [comment]
Yep - HttpClient stays in a Maui app because of these 2 places:

![image](https://user-images.githubusercontent.com/8291187/151382973-676b5888-16e6-4de8-b481-4613956498d3.png)

**Comment by eerhardt** [comment]
I just tested a prototype fix for this. On a `dotnet new maui` app with build options:

```
dotnet build -c Release -t:Run -f net6.0-android -r android-x64 -p:RunAOTCompilation=true -p:AndroidEnableProfiledAot=true
```

Fixing this takes the profiled AOT .apk from

Before: `14.1 MB (14,787,482 bytes)`
After: `13.3 MB (14,040,836 bytes)`

Note: Profiled AOT is enabled by default in the latest Xamarin-Android SDK, so I'm just simulating the latest default Release build here. See https://github.com/xamarin/xamarin-android/pull/6547

**Comment by eerhardt** [comment]
This was fixed in the `release/6.0` branch. Closing.
